### DISCUSSION TITLE
How to justify content with display flex, if my StyledForm wont recognise my inputbars?

### DISCUSSION THREAD

**Discussion by PhilipZi** [neuefische/questions#82] [Web] [answered]
Hallo Zusammen, 
ich versuche meinen content in meiner StyledForm auszurichten allerdings funktioniert es nicht, da die Styledform die input Elemente nicht ganz abbildet (siehe Bild) Weiss jemand was da los ist? also mit display flex und justify content passiert da gar nichts, tut sich nichts. 
![Screenshot 2022-08-18 at 14 25 33](https://user-images.githubusercontent.com/102238180/185395267-6fadfc5c-cd0b-4332-9311-9151c964cb39.png)

**Comment by PReichetanz** [comment]
Hallo,

`display: flex` macht immer nur die direkten Kindelemente zu Flex-Items; in Deinem Fall sind das die drei `divs`. In dem zweiten dieser `divs` befinden sich Deine Labels und die Inputelemente, also versuche mal, diesem `div` ebenfalls `display: flex` zu geben.

**Reply by PhilipZi** [reply]
Danke für die Antwort, aber es geht in dem fall nicht um die Inputelemente sondern um den Checkmark button der in dem dritten div ist und dieser wird zwischen den input Feldern angezeigt :( das ist das Problem welches ich habe. Ich glaube er rendert die Inputfelder nicht als Vertical aus sonder als horizontal aber ich habe keine Ahnung wie ich das lösen könnte.

**Reply by PReichetanz** [reply]
Dann bitte einmal (so wie eigentlich immer) den aktuellen Stand pushen und den Link zu dem Branch / der betroffenen Datei hier angeben, damit wir es uns anschauen können.

**Reply by PhilipZi** [reply]
Kein Problem, habe ich aus versehen vergessen. Hier der link zu meinem Git Hub branch
https://github.com/PhilipZi/capstone-project/blob/edit-timer-card/src/components/timer/timer1.js

Vielen Dank fürs rüber schauen!

**Reply by ahohnsen** [reply]
Verstehe ich es richtig, dass du den Checkmark-Button einfach unter den Inputfeldern anzeigen lassen möchtest anstatt dazwischen?

**Reply by PhilipZi** [reply]
ja genau das ist correct

**Reply by PhilipZi** [reply]
der grüne bg hier zieht sich über den gesamten body also irgendwas ist ganz komisch hier. 
![Screenshot 2022-08-18 at 16 06 33](https://user-images.githubusercontent.com/102238180/185415416-bf7d84e2-6c38-4cec-9814-d7d2db56f71d.png)

**Comment by alexclaes** [comment, marked as answer]
Wenn ich richtig verstehe, was du hier beschreibst und du mir vorhin erklärst hast, sind das ja zwei Probleme.

**1.) Der grüne Bereich soll über die gesamte Höhe des Viewports gehen (so ist es ja auch in dem Layout in deiner User Story)**

Grundsätzlich ist jedes Block Element erst einmal nur so hoch, wie sein Inhalt - außer du nimmst per CSS Einfluss darauf.

Das macht du auch bei bei dem `main` Element, indem du die Property `height` setzt. Dieses Element benötigt dann auch die Property `background-color`. Du setzt die Hintergrundfarbe jedoch auf einem darin verschachtelten `div` Element.

**2.) Anordnung von Slidern und Buttons passt nicht.**

Es soll etwa so aussehen, oder?

<img src="https://user-images.githubusercontent.com/7231474/185438039-2a0a6a2e-da3d-478d-9765-8941bc4c2480.png" width="300" />

Die Input Felder sind um 90 Grad gedreht und verschoben, weil du das in deinem CSS Code explizit so programmiert hast. 

Bitte überprüfe die Styled Component für die Input Felder und entferne die nicht benötigten Angaben.

**Reply by PhilipZi** [reply]
Vielen Dank Alex, das hat mir sehr geholfen. War echt am verzweifeln.
